src/test_l1_plat_power.c

```diff
@@ -482,42 +484,25 @@ void test_l1_plat_power_positive_PLAT_API_SetWakeupSrc (void)
 {
     gTestID = 9;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
-	pmStatus_t ret;
+    pmStatus_t ret;
+    PWRMGR_WakeupSrcType_t srcType = PWRMGR_WAKEUPSRC_MAX;
+    char keyvalue[POWER_MANAGER_KEY_SIZE] = {0};
+    uint8_t countOfWakeupSources = 0;
 
     // Variation/Step 01: Call PLAT_INIT() - open interface
     ret = PLAT_INIT();
     UT_ASSERT_EQUAL(ret, PWRMGR_SUCCESS); // Ensure the returned value is PWRMGR_SUCCESS
 
-    typedef struct {
-        PWRMGR_WakeupSrcType_t src;
-        bool enabled;
-    } WakeupTest;
-
-    WakeupTest tests[] = {
-        { PWRMGR_WAKEUPSRC_VOICE, false },
-        { PWRMGR_WAKEUPSRC_VOICE, true },
-        { PWRMGR_WAKEUPSRC_PRESENCE_DETECTION, false },
-        { PWRMGR_WAKEUPSRC_PRESENCE_DETECTION, true },
-        { PWRMGR_WAKEUPSRC_BLUETOOTH, false },
-        { PWRMGR_WAKEUPSRC_BLUETOOTH, true },
-        { PWRMGR_WAKEUPSRC_WIFI, false },
-        { PWRMGR_WAKEUPSRC_WIFI, true },
-        { PWRMGR_WAKEUPSRC_IR, false },
-        { PWRMGR_WAKEUPSRC_IR, true },
-        { PWRMGR_WAKEUPSRC_POWER_KEY, false },
-        { PWRMGR_WAKEUPSRC_POWER_KEY, true },
-        { PWRMGR_WAKEUPSRC_TIMER, false },
-        { PWRMGR_WAKEUPSRC_TIMER, true },
-        { PWRMGR_WAKEUPSRC_CEC, false },
-        { PWRMGR_WAKEUPSRC_CEC, true },
-        { PWRMGR_WAKEUPSRC_LAN, false },
-        { PWRMGR_WAKEUPSRC_LAN,true }
-    };
-
-    // Variation/Step 2-19: Verify that the function runs properly with each possible input
-    for(int i = 0; i < (sizeof(tests)/sizeof(tests[0])); i++) {
-        UT_LOG("\n Function: %s Wake up source test inputs values :[%d%d]\n", __FUNCTION__, tests[i].src, tests[i].enabled);
-        ret = PLAT_API_SetWakeupSrc(tests[i].src, tests[i].enabled);
+    // Variation/Step 02:Getting the count of the supported wakeup sources
+    countOfWakeupSources = UT_KVP_PROFILE_GET_UINT32("powermanager/CountOfWakeupSources");
+    UT_LOG_DEBUG ("Count Of Wakeup Sources %d" , countOfWakeupSources);
+
+    // Variation/Step 03:Getting the supported wakeup sources
+    for(int i = 0; i < countOfWakeupSources; i++){
+        snprintf(keyvalue, POWER_MANAGER_KEY_SIZE, "powermanager/WakeupSources/%d", i);
+        srcType = UT_KVP_PROFILE_GET_UINT32(keyvalue);
+	ret = PLAT_API_SetWakeupSrc(srcType, true);
+        UT_LOG_DEBUG("Return status: %d", ret);
```

Comment:
have not added code for step3 PLAT_API_SetWakeupSrc(srcType, false) case

Reply:
Done